src/custom-blocks/auto-item-list/index.php

```diff
@@ -66,50 +95,104 @@ function render_callback_auto_item_list_block($attributes)
             // Restore original Post Data
             wp_reset_postdata();  
 
+            $number_of_items = count($item_array);
+            $max_number_of_items = 3;
+            $few_items_class = "";
+            $image_size = "medium";
+            if ($number_of_items < $max_number_of_items) {
+                $few_items_class = " mojblocks-auto-item-list__item--$number_of_items";
+            }
     ?>
 
     <div class="<?php _e(esc_html($attribute_box_className)); ?> mojblocks-auto-item-list">
-        <div class="govuk-width-container govuk-!-margin-0">
-            <div class="govuk-grid-row">
-                <?php
-                    $i = 0;
-                    $number_of_items = count($item_array);
-                    if ($number_of_items) {
-                        while ($i < $number_of_items && $i < 3) {
-                ?>
-                            <div class="mojblocks-auto-item-list__item">
-                                <p class="govuk-body mojblocks-auto-item-list__headline" >
-                                    <?php 
+        <?php
+            $i = 0;
+            if ($number_of_items) {
+                if ($attribute_box_listImage) {
+                    //If images are enabled, we need to check whether all/any items have images
+                    $all_items_have_images = true; // assume all have images
+                    $one_items_has_image = false; // assume none have an image
+                    for ($j=0; $j < $number_of_items && $j < $max_number_of_items; $j++) {
+                        // check that all items have an image
+                        $id = $item_array[$j]['id'];
+                        if (!has_post_thumbnail($id)) {
+                            $all_items_have_images = false;
+                        } else {
+                            $one_items_has_image = true;
+                            $has_image_class = "has-image";
+                        }
+                    }
+                }
+                while ($i < $number_of_items && $i < $max_number_of_items) {
+                    $id           = $item_array[$i]["id"];
+                    $backup_image = !empty($attribute_box_listBackupImage) ? $attribute_box_listBackupImage : "";
+                    $image        = has_post_thumbnail($id) ? wp_get_attachment_image_src(get_post_thumbnail_id($id),$image_size)[0] : $backup_image;
+                    $title        = __(esc_html($item_array[$i]["title"]),"hale");
+                    $summary      = __(esc_html($item_array[$i]["summary"]),"hale");
+                    $date         = $attribute_box_hasDate ? date(get_option("date_format"), strtotime($item_array[$i]["date"])) : "";
+                    $url          = esc_html($item_array[$i]["link"]);
```

Comment:
If it is just a URL I think you could use `esc_url()` ?

Reply:
Thanks, changed